src/controllers/organization.ts

```diff
@@ -106,22 +109,42 @@ export const postJoinOrganizationMiddleware = async (
     });
 
     const { confirmed, siret } = await schema.parseAsync(req.body);
+    const { id: user_id } = getUserFromAuthenticatedSession(req);
+
+    if (req.session.certificationDirigeantRequested) {
```

Comment:
suggestion: the joinOrganization function is complex already. We should keep the join login in one place rather that using extending the logic in this controller.